main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	mapset "github.com/deckarep/golang-set"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+)
+
+func main() {
+	start()
+}
+
+func start() {
+	f, err := os.Open("TestExpectations.txt")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		expectation := scanner.Text()
+		url, tags, testname, results := parseExpectation(expectation)
+		if isMeetCriteria(url, tags, testname, results) {
+			bsfSet := getBSF()
+			if bsfSet.Contains(testname) {
+				fmt.Println(expectation)
+			}
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// https://github.com/web-platform-tests/wpt.fyi/issues/1993#issuecomment-654551961
+func isMeetCriteria(url, tags, testname, results string) bool {
+	if url == "" {
+		return false
+	}
+
+	if testname == "" {
+		return false
+	}
+
+	if results != "Timeout" && results != "Failure" {
+		return false
+	}
+
+	if tags != "" {
+		if !strings.Contains(tags, "Linux") {
+			if tags != "Release" {
+				return false
+			}
+		} else {
+			if strings.Contains(tags, "Debug") {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// syntax: https://chromium.googlesource.com/chromium/src/+/master/docs/testing/web_test_expectations.md#syntax
+// [ bugs ] [ "[" modifiers "]" ] test_name [ "[" expectations "]" ]
+func parseExpectation(line string) (url string, tags string, testname string, results string) {
+	if strings.HasPrefix(line, "# ") {
+		return
+	}
+
+	split := strings.Split(line, " ")
+	//fmt.Println("new line")
+	//fmt.Println(split)
+	for i := 0; i < len(split); i++ {
+		if strings.HasPrefix(split[i], "crbug.com") {
+			url = split[i]
+			//fmt.Println(url)
+		}
+
+		if split[i] == "[" {
+			val := ""
+			i++
+			for split[i] != "]" {
+				val += split[i]
+				i++
+			}
```

Comment:
This doesn't matter for this script's purpose, but just noting that if this were real code `val` should be an array that we accumulate, rather than squashing strings together.